symfony/plugins/orangehrmAdminPlugin/Api/ValidationUserNameAPI.php

```diff
@@ -0,0 +1,108 @@
+<?php
+/**
+ * OrangeHRM is a comprehensive Human Resource Management (HRM) System that captures
+ * all the essential functionalities required for any enterprise.
+ * Copyright (C) 2006 OrangeHRM Inc., http://www.orangehrm.com
+ *
+ * OrangeHRM is free software; you can redistribute it and/or modify it under the terms of
+ * the GNU General Public License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Admin\Api;
+
+use OrangeHRM\Admin\Service\UserService;
+use OrangeHRM\Core\Api\V2\Endpoint;
+use OrangeHRM\Core\Api\V2\EndpointResourceResult;
+use OrangeHRM\Core\Api\V2\EndpointResult;
+use OrangeHRM\Core\Api\V2\Model\ArrayModel;
+use OrangeHRM\Core\Api\V2\RequestParams;
+use OrangeHRM\Core\Api\V2\ResourceEndpoint;
+use OrangeHRM\Core\Api\V2\Validator\ParamRule;
+use OrangeHRM\Core\Api\V2\Validator\ParamRuleCollection;
+use OrangeHRM\Core\Api\V2\Validator\Rule;
+use OrangeHRM\Core\Api\V2\Validator\Rules;
+use OrangeHRM\Core\Traits\ServiceContainerTrait;
+use OrangeHRM\Framework\Services;
+
+class ValidationUserNameAPI extends Endpoint implements ResourceEndpoint
+{
+    use ServiceContainerTrait;
+
+    public const PARAMETER_USER_NAME = 'userName';
+    public const PARAMETER_IS_CHANGEABLE_USERNAME = 'valid';
+
+    /**
+     * @return UserService|null
+     */
+    public function getSystemUserService(): ?UserService
+    {
+        return $this->getContainer()->get(Services::USER_SERVICE);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getOne(): EndpointResult
+    {
+        $userName = $this->getRequestParams()->getString(RequestParams::PARAM_TYPE_QUERY, self::PARAMETER_USER_NAME);
+        $isChangeableUserName = $this->getSystemUserService()->getSystemUserDao()->isUserNameExistByUserName($userName);
+        return new EndpointResourceResult(
+            ArrayModel::class,
+            [
+                self::PARAMETER_IS_CHANGEABLE_USERNAME => $isChangeableUserName,
+                self::PARAMETER_USER_NAME => $userName,
```

Comment:
will remove this. no point of adding request parameter to the response. (remove const as well)

Reply:
done